### PR TITLE
Ensure products rebuild when modules are modified

### PR DESCRIPTION
**Pull request by mxcl**
Fixes SR-1136

/cc @ddunbar @aciidb0mb3r

### CODE REVIEW

**Comment by mxcl** [comment]
@swift-ci Please test

/cc @ddunbar @aciidb0mb3r

**Comment by mxcl** [comment]
@swift-ci Please test

File: Sources/Build/Command.link().swift [hunk 1]
```diff
@@ -46,16 +46,7 @@ extension Command {
...
-            switch conf {
-            case .Debug:
-                let tool = SwiftcTool(module: module, prefix: prefix, otherArgs: [], executable: SWIFT_EXEC)
-                // must return tool’s outputs and inputs as shell nodes don't calculate more than that
-                return tool.inputs + tool.outputs
-            case .Release:
-                return objects
-            }
-        }
+        let inputs = objects
```
Comment (aciidb0mb3r) [inline Sources/Build/Command.link().swift]: nitpick: don't really need the local variable anymore

Comment (mxcl) [inline Sources/Build/Command.link().swift]: True

**Comment by mxcl** [comment]
I merged even though OS X failed, because it failed in some non-swiftpm tests and I tested it locally and I'm willing to take the blame if it is not in fact good.

**Comment by aciidb0mb3r** [comment]
we should also put some testcase for this

File: Sources/Build/Command.link().swift (hunk 1 above)
Comment (ddunbar) [inline Sources/Build/Command.link().swift]: There should also probably be dependencies to the other ancillary outputs of the Swift build, e.g. the .swiftmodule files.

Comment (mxcl) [inline Sources/Build/Command.link().swift]: Yeah, I'll try to do a more thorough job here.

**Comment by mxcl** [comment]
Yes, we do need a test case for it. I find myself reluctant to add additional functional tests and first want to clean all this up, but this was wrong of me.

**Pull request merged**